### PR TITLE
Add react example

### PR DESCRIPTION
**Pull request by alejandronanez**
- Add React example under /examples

### CODE REVIEW

**Comment by arianon** [comment]
I think you should be using [poi-preset-react](https://github.com/egoist/poi/tree/master/packages/poi-preset-react) instead of manually configuring Babel.

**Comment by alejandronanez** [comment]
Hey @arianon, that didn't work the way I expected, by updating `.babelrc` with the poi preset I got this error:

```js
// .babelrc
{
  "presets": [
    "poi-preset-react"
  ]
}
```

```shell
error: Unsupported preset format: poi => {
    poi.extendWebpack(config => {
      // Add react plugins to babel config
      config.module.rule('js')
        .use('babel-loader')
        .tap(options => {
          // Remove default vue-app preset
          // When `babelrc` is false, we're using default babel config
          if (options.babelrc === false) {
            options.presets = []
          }
          return poi.merge(options, {
            presets: [
              require.resolve('babel-preset-react-app')
            ],
            plugins: [
              require.resolve('babel-plugin-react-require'),
              require.resolve('react-hot-loader/babel')
            ]
          })
        })

      // Add entry
      if (config.entryPoints.has('client')) {
        config.entry('client').prepend(require.resolve('react-hot-loader/patch'))
      } else {
        throw new Error('Currently only `client` entry is supported')
      }
    })
  }. (While processing preset: "/Users/alejandro/code/poi/examples/react/node_modules/poi-preset-react/index.js")
    at /Users/alejandro/code/poi/examples/react/node_modules/babel-core/lib/transformation/file/options/option-manager.js:320:17
    at Array.map (<anonymous>)
    at OptionManager.resolvePresets (/Users/alejandro/code/poi/examples/react/node_modules/babel-core/lib/transformation/file/options/option-manager.js:275:20)
    at OptionManager.mergePresets (/Users/alejandro/code/poi/examples/react/node_modules/babel-core/lib/transformation/file/options/option-manager.js:264:10)
    at OptionManager.mergeOptions (/Users/alejandro/code/poi/examples/react/node_modules/babel-core/lib/transformation/file/options/option-manager.js:249:14)
    at OptionManager.init (/Users/alejandro/code/poi/examples/react/node_modules/babel-core/lib/transformation/file/options/option-manager.js:368:12)
    at File.initOptions (/Users/alejandro/code/poi/examples/react/node_modules/babel-core/lib/transformation/file/index.js:212:65)
    at new File (/Users/alejandro/code/poi/examples/react/node_modules/babel-core/lib/transformation/file/index.js:135:24)
    at Pipeline.transform (/Users/alejandro/code/poi/examples/react/node_modules/babel-core/lib/transformation/pipeline.js:46:16)
    at transpile (/Users/alejandro/code/poi/examples/react/node_modules/babel-loader/lib/index.js:50:20)
    at /Users/alejandro/code/poi/examples/react/node_modules/babel-loader/lib/fs-cache.js:118:18
    at ReadFileContext.callback (/Users/alejandro/code/poi/examples/react/node_modules/babel-loader/lib/fs-cache.js:31:21)
    at FSReqWrap.readFileAfterOpen [as oncomplete] (fs.js:420:13)
```

**Comment by arianon** [comment]
`poi-preset-react` doesn't belong in the `.babelrc` (it's a Poi preset, not a Babel one, but I understand the confusion) it belongs in the `poi.config.js` and is used like so

```js
module.exports = {
  presets: [
    require('poi-preset-react')()
  ]
}
```

**Comment by egoist** [comment]
Can you update example to make hot reloading work too? similar to https://github.com/egoist/poi/tree/master/packages/poi-preset-react/example

**Comment by alejandronanez** [comment]
Hey @egoist, sure will do.

File: examples/react/package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
   },
   "dependencies": {
     "react": "^16.0.0",
-    "react-dom": "^16.0.0"
+    "react-dom": "^16.0.0",
+    "react-hot-loader": "^3.1.1"
```
Comment (egoist) [inline examples/react/package.json]: no need of this, we had it in `poi-preset-react` already

**Comment by egoist** [comment]
one last thing, let's remove all those semicolons 😄

**Pull request merged**